src/mainboard/canaan/canmv-k230/bt0/src/util.rs

```diff
@@ -0,0 +1,35 @@
+use core::ptr::{read_volatile, write_volatile};
+use core::slice;
+
+use log::{print, println};
+
+pub fn udelay(time: usize) {
+    for _ in 0..time {
+        unsafe { core::arch::asm!("nop") }
+    }
+}
+
+pub fn dump(addr: usize, length: usize) {
```

Comment:
This kind of thing is going to become code mess at some point, is there no way to put it in a common crate?

Reply:
I have that on my agenda, more to come.